### PR TITLE
doc: add LGPLv3 license

### LINKED ISSUE #165: add `LICENSE`
LGPLv3?

### PR DESCRIPTION
**Pull request by c-dilks**
close #165 

1. add `LICENSE`
1. add copyright notice comments on all source code, including author names
1. add `SPDX` file
1. add notice about `subrepo` licenses to `deps/README.md`

### CODE REVIEW

**Comment by c-dilks** [comment]
@duanebyer I asked all contributors about this license choice via email, but I'm not sure I have your updated email address. Are you okay with this choice of license?

**Comment by c-dilks** [comment]
> 
>     1. add `LICENSE`
> 
>     2. add copyright notice comments on all source code, including author names
> 
>     3. add `SPDX` file
> 
>     4. add notice about `subrepo` licenses to `deps/README.md`

@wdconinc does this PR look okay to you? I checked 1 and 3 with online SPDX validators, and I checked 2 carefully. Number 4 I'm concerned about, since we use [git subrepos](https://github.com/ingydotnet/git-subrepo) to manage a few dependencies as mirrors in subdirectories of `deps/`; I can't find any suggested way to deal with this situation regarding licenses. Alternatively, we could put these dependencies in `eic-shell` but I don't think that's necessary at this time.

File: deps/README.md [hunk 1]
```diff
@@ -16,6 +16,10 @@ while the others must be downloaded separately.
 - `adage`, from [c-dilks/adage](https://github.com/c-dilks/adage):
   analysis framework for multi-dimensional binning
 
+Notice
+======
+These dependencies (subdirectories) may have a different license than `sidis-eic` itself. Since they are git subrepos, they have been
+copied as is from their original source; the license and copyright used by `sidis-eic` does not apply to these subdirectories.
```
Comment (c-dilks) [inline deps/README.md:23]: Here is the notice

**Comment by wdconinc** [comment]
For 3rdparty included code licensing, see e.g. https://github.com/qt/qtbase/tree/dev/src/3rdparty

**Comment by c-dilks** [comment]
I updated the 3rd party notice a bit:
```
These dependencies (subdirectories) may have a different license than
`sidis-eic` itself. The license and copyright used by `sidis-eic` does not
apply to these subdirectories. Since they are git subrepos, they have been
copied as is from their original source. Changes to any code in these
dependencies must be accepted in the corresponding public Git repositories
before being mirrored here.
```
I will merge this PR soon; if we need to change this notice we can handle it with a separate PR.

**Comment by c-dilks** [comment]
@duanebyer this is merged now. I added copyright notices on all source code, let me know if you find any mistakes for your contributions.

**Pull request merged**